keystore/keystore.go

```diff
@@ -149,5 +153,19 @@ func (ks *FSKeystore) List() ([]string, error) {
 		return nil, err
 	}
 
-	return dir.Readdirnames(0)
+	dirs, err := dir.Readdirnames(0)
+	if err != nil {
+		return nil, err
+	}
+
+	var list []string
```

Comment:
Probably best to preallocate `list := make([]string, 0, len(dirs))`.

Reply:
I'd leave out the capacity param since we don't know if all the dirs are gonna make it to the list.

Reply:
Over allocating slightly is better than under allocating, in general (especially when we only over allocate in an error case). Probably not that much of an issue here as this function isn't frequently called but I know it'll continue to bug me.

Reply:
done 👍